integration-tests/stax/src/main/java/org/apache/camel/quarkus/component/stax/it/StaxResource.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.stax.it;
+
+import java.io.InputStream;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.quarkus.component.stax.it.model.Record;
+
+@Path("/stax")
+public class StaxResource {
+
+    @Inject
+    CamelContext context;
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Path("/records")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public int staxProcessRecords() throws Exception {
+        InputStream resource = StaxResource.class.getResourceAsStream("/records.xml");
+        TagCountHandler countingHandler = producerTemplate.requestBody("direct:processRecords", resource,
+                TagCountHandler.class);
+        return countingHandler.getCount();
```

Comment:
Is there any reason not to `try(InputStream resource = StaxResource.class.getResourceAsStream("/records.xml"))` here and elsewhere in this class?

Reply:
Good catch - I'll change that.